lib/spack/spack/build_systems/python.py

```diff
@@ -233,7 +233,15 @@ def install_args(self, spec, prefix):
         if ('py-setuptools' == spec.name or          # this is setuptools, or
             'py-setuptools' in spec._dependencies and  # it's an immediate dep
             'build' in spec._dependencies['py-setuptools'].deptypes):
-            args += ['--single-version-externally-managed', '--root=/']
+            args += ['--single-version-externally-managed']
+
+        python_pkg = spec['python'].package
+        args += ['--root=%s' % prefix,
+                 '--install-lib=%s' % python_pkg.site_packages_dir,
```

Comment:
On some systems, Python libraries are automatically installed to `prefix.lib64`. This will force them to install to `prefix.lib`. I'm not sure if that will break things or not, but it's something to be aware of.

Reply:
@adamjstewart do you know which systems? I'll spin up a container and see if I can break things.

Reply:
I've seen reports of this on opensuse, fedora, and rhel: #18520, #17126, #19546

Reply:
Ok, so I think what we really need to be using is `distutils.sysconfig.get_python_lib(True, prefix=prefix)`. That will give the platform-specific directory (which is generally used for compiled code) which is the more restrictive.

Reply:
In that case, should we make the attribute in the Python package more robust?

Reply:
Ok, I think this is fixed now, and handled "properly"